cmd/api/src/auth/role_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2023 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build integration
+// +build integration
+
+package auth_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gofrs/uuid"
+	"github.com/specterops/bloodhound/lab"
+	"github.com/specterops/bloodhound/src/api"
+	v2 "github.com/specterops/bloodhound/src/api/v2"
+	"github.com/specterops/bloodhound/src/auth"
+	"github.com/specterops/bloodhound/src/model"
+	"github.com/specterops/bloodhound/src/model/appcfg"
+	"github.com/specterops/bloodhound/src/test/lab/fixtures"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func testCondition(role auth.RoleTemplate, permission model.Permission) string {
+	if role.Permissions.Has(permission) {
+		return "SHOULD"
+	}
+	return "SHOULD NOT"
+}
+
+func requireForbidden(t *testing.T, err error) {
```

Comment:
Replace `t *testing.T` with `assert *require.Assertions` so that you are not mixing up testing contexts